inspect_image.go

```diff
@@ -65,9 +72,28 @@ func (c *Client) InspectImage(name string, daemon bool) (*ImageInfo, error) {
 		return nil, err
 	}
 
-	defaultProcessType, err := img.Env("CNB_PROCESS_TYPE")
-	if err != nil || defaultProcessType == "" {
-		defaultProcessType = "web"
+	platformAPI, err := img.Env(PlatformAPIEnv)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading platform api")
+	}
+
+	if platformAPI == "" {
+		platformAPI = fallbackPlatformAPI
+	}
+
+	platformAPIVersion, err := semver.NewVersion(platformAPI)
+	if err != nil {
+		return nil, errors.Wrap(err, "parsing platform api version")
+	}
+
+	var defaultProcessType string
+	if platformAPIVersion.LessThan(semver.MustParse("0.4")) {
+		defaultProcessType, err = img.Env(CNBProcessEnv)
+		if err != nil || defaultProcessType == "" {
+			defaultProcessType = defaultProcess
+		}
+	} else {
+		defaultProcessType = defaultProcess
```

Comment:
to be fair, I'm not sure on this either. I just don't see in the spec how to know what process type was set, outside of the `web` default.